token/rust/src/client.rs

```diff
@@ -0,0 +1,235 @@
+use async_trait::async_trait;
+use solana_client::rpc_client::RpcClient;
+use solana_program_test::{tokio::sync::Mutex, BanksClient, ProgramTestContext};
+use solana_sdk::{
+    account::Account, hash::Hash, pubkey::Pubkey, signature::Signature, transaction::Transaction,
+};
+use std::{fmt, future::Future, pin::Pin, sync::Arc};
+
+type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T> + Send + 'a>>;
+
+/// Basic trait for sending transactions to validator.
+pub trait SendTransaction {
+    type Output;
+}
+
+/// Extends basic `SendTransaction` trait with function `send` where client is `&mut BanksClient`.
+/// Required for `TokenBanksClient`.
+pub trait SendTransactionBanksClient: SendTransaction {
+    fn send<'a>(
+        &self,
+        client: &'a mut BanksClient,
+        transaction: Transaction,
+    ) -> BoxFuture<'a, TokenClientResult<Self::Output>>;
+}
+
+/// Send transaction to validator using `BanksClient::process_transaction`.
+#[derive(Debug, Clone, Copy, Default)]
+pub struct TokenBanksClientProcessTransaction;
+
+impl SendTransaction for TokenBanksClientProcessTransaction {
+    type Output = ();
+}
+
+impl SendTransactionBanksClient for TokenBanksClientProcessTransaction {
+    fn send<'a>(
+        &self,
+        client: &'a mut BanksClient,
+        transaction: Transaction,
+    ) -> BoxFuture<'a, TokenClientResult<Self::Output>> {
+        Box::pin(async move {
+            client
+                .process_transaction(transaction)
```

Comment:
It's a pity that the BanksClient method(s) don't take &Transaction.
Can you pass by reference until here, though?

Reply:
Pass by reference and create `Transaction` with clone for BanksClient method?

Reply:
Yes, that's what I was thinking. I don't feel strongly about it, though

Reply:
Changed to reference.